### PR TITLE
Admin

### PR DESCRIPTION
**Pull request by molinajulian**
## Summary
The creation of users with administrator permissions is added.

## Trellocard
https://trello.com/c/dbpT54ct

### CODE REVIEW

File: Training_node/first_project/app/controllers/users.js [hunk 1]
```diff
@@ -81,18 +77,15 @@ exports.create = (req, res, next) => {
...
-                })
-                .catch(err => {
-                  next(err);
-                });
+            if (admin) {
+              user.admin = true;
+              return exist ? setAdmin(exist, res, next) : createModel(user, res, next);
             } else {
-              next(errors.existsUser);
+              if (!exist) {
```
Comment (gonziesc) [inline Training_node/first_project/app/controllers/users.js]: you dont need this

File: Training_node/first_project/app/controllers/users.js [hunk 2]
```diff
@@ -107,6 +100,43 @@ exports.create = (req, res, next) => {
...
+      } else {
+        next(errors.incorrect_user_password);
+      }
+    });
+  } else {
+    next(errors.defaultError(result.reason));
+  }
+};
+exports.createAdmin = (req, res, next) => {
+  return create(req, res, next, true);
```
Comment (gonziesc) [inline Training_node/first_project/app/controllers/users.js]: awesome

File: Training_node/first_project/app/middlewares/auth.js [hunk 3]
```diff
@@ -1,16 +1,26 @@
...
+const auth = (req, res, next, isAdmin = false) => {
+  const authHeader = req.headers[sessionManager.HEADER_NAME];
+  if (authHeader) {
+    const user = sessionManager.decode(authHeader);
     return User.getByEmail(user)
       .then(u => {
         if (u) {
-          req.user = u;
-          next();
+          if (isAdmin) {
```
Comment (gonziesc) [inline Training_node/first_project/app/middlewares/auth.js]: why this two ifs?

File: Training_node/first_project/app/controllers/users.js (hunk 2 above)
Comment (molinajulian) [reply Training_node/first_project/app/controllers/users.js]: Thanks! :smile:

File: Training_node/first_project/app/controllers/users.js (hunk 1 above)
Comment (molinajulian) [reply Training_node/first_project/app/controllers/users.js]: Why ? If the user exist and you do not want to create an admin, it is an error

File: Training_node/first_project/app/middlewares/auth.js (hunk 3 above)
Comment (molinajulian) [reply Training_node/first_project/app/middlewares/auth.js]: It's similar create(createAdmin and createRegular) but in this context. If you want verify the log-in user regular, if the exist there is enough but if you want verify the log-in admin also verify if the user have permissions.

**Pull request merged**